### PR TITLE
Use default cache dir if there are errors with configured cache dir

### PR DESCRIPTION
**Pull request by shanejbrown**
## Description

Use the default cache directory if there are any errors with the cache root dir in the global config. Rather than failing the build, this will print warnings of the failures with the directory and proceed with the building using the default cache directory on the local file system.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by shanejbrown** [comment]
Examples of warning messages:
`2023-12-12 14:36:25,272 buildrunner                    WARNING  There was a an issue with /mnt/caches/buildrunner-bogus: [Errno 13] Permission denied: '/mnt/caches/buildrunner-bogus'
2023-12-12 14:36:25,273 buildrunner                    WARNING  Using ~/.buildrunner/caches for cache directory`

File: buildrunner/__init__.py [hunk 1]
```diff
@@ -419,21 +419,29 @@ def get_cache_archive_file(self, cache_name, project_name=""):
...
-        cache_dir = os.path.dirname(local_cache_archive_file)
-        if not os.path.exists(cache_dir):
-            os.makedirs(cache_dir)
+        caches_root = self.global_config.get('caches-root', DEFAULT_CACHES_ROOT)
+        local_cache_archive_file = None
+        try:
+            local_cache_archive_file = get_filename(caches_root, cache_name)
+        except Exception as exc:  # pylint: disable=broad-except
+            # Intentinally catch all exceptions here since we don't want to fail the build
+            LOGGER.warning(f'There was a an issue with {caches_root}: {str(exc)}')
```
Comment (bluesliverx) [inline buildrunner/__init__.py:441]: "a an" -> "an"
I would also call out that this is about caching to that dir, so we know why it wants to use that dir

File: buildrunner/__init__.py [hunk 2]
```diff
@@ -419,21 +419,29 @@ def get_cache_archive_file(self, cache_name, project_name=""):
         """
         Given a cache name determine the local file path.
         """
-        caches_root = self.global_config.get('caches-root', DEFAULT_CACHES_ROOT)
-        build_path = os.path.splitdrive(self.build_dir)[1]
-        if os.path.isabs(build_path):
-            build_path = build_path[1:]
+        def get_filename(caches_root, cache_name):
```
Comment (bluesliverx) [inline buildrunner/__init__.py:422]: This is only used one place, why have a sub-method here?

Comment (shanejbrown) [reply buildrunner/__init__.py:422]: It's used two places in the `try` and in the `except`. I just didn't want to duplicate the code.

File: buildrunner/__init__.py (hunk 1 above)
Comment (shanejbrown) [reply buildrunner/__init__.py:441]: Oops! Nice catch. I'll fix that.

File: buildrunner/__init__.py (hunk 2 above)
Comment (bluesliverx) [reply buildrunner/__init__.py:422]: browser search, you have failed me for the last time...

**Review by bluesliverx — APPROVED** [review]

**Pull request merged**